File: server/src/main/java/com/ufit/server/service/ReportService.java
package com.ufit.server.service;

import java.util.List;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import com.ufit.server.dto.response.ReportDto;
import com.ufit.server.entity.Report.ReportStatus;

public interface ReportService {
    void reportPost(Long postId, String reason, String reporterUsername);
    void reportArticle(Long articleId, String reason, String reporterUsername);
    
    // Get paginated reports with optional status filter
    Page<ReportDto> getReportsForMod(ReportStatus status, Pageable pageable);
    Page<ReportDto> getReportsForAdmin(ReportStatus status, Pageable pageable);
    
    // Get pending reports (for backward compatibility)
    List<ReportDto> getPendingReportsForMod();
    List<ReportDto> getPendingReportsForAdmin();
    
    void reviewReportAsMod(Long reportId, String action);
    void reviewReportAsAdmin(Long reportId, String action);
}

File: frontend/src/services/reportService.js
// src/services/reportService.js
import { API } from "./api";

/**
 * Lấy tất cả report có status = PENDING từ backend cho cả mod và admin.
 * - Với mod: gọi /api/mod/reports
 * - Với admin: gọi /api/admin/reports
 * Ở đây chỉ cung cấp hàm hỗ trợ chung, nếu cần phân biệt thì bạn có thể tách thành 2 hàm.
 */

function getPendingReportsForMod() {
	// GET /api/mod/reports
	return API.get("/api/mod/reports");
}

function getPendingReportsForAdmin() {
	// GET /api/admin/reports
	return API.get("/api/admin/reports");
}

function reviewReportAsMod(reportId, action) {
	// PUT /api/mod/reports/{reportId}/review?action=…
	return API.put(`/api/mod/reports/${reportId}/review`, null, {
		params: { action },
	});
}

function reviewReportAsAdmin(reportId, action) {
	// PUT /api/admin/reports/{reportId}/review?action=…
	return API.put(`/api/admin/reports/${reportId}/review`, null, {
		params: { action },
	});
}

const ReportService = {
	getPendingReportsForMod,
	getPendingReportsForAdmin,
	reviewReportAsMod,
	reviewReportAsAdmin,
};

export default ReportService;
